### PR TITLE
Refactor/log viewer with hooks

### PR DESCRIPTION
**Pull request by mewhhaha**
**EDIT** *This is a suggestion of a refactor! We don't have to use hooks, but if you think it's nicer and cleaner, then I think we should!*

This pull request will refactor mostly LogViewerList and LogViewer.

LogViewerList has been reduced by about 50 LoC and I think it's more clear what's happening in the component now, so that it's easier to change things in the future. 

The previous cache files have been replaced by a custom hook called useCache. Some small syntax refactoring in reducers, and a way in the menu reducer to see what the currently opened file is by adding a parameter called currentSourceHandle.

Functionality hasn't changed.

### CODE REVIEW

File: src/js/view/components/LogViewer.js [hunk 1]
```diff
@@ -8,43 +8,40 @@ import { connect } from 'react-redux';
...
 
-class LogViewer extends React.Component {
-  constructor(props) {
-    super(props);
+const LogViewer = props => {
+  const lines = props.logs[props.source.path];
 
-    this.windowedList = React.createRef();
-  }
+  const [highlightRegExp, setHighlightRegExp] = useState(undefined);
```
Comment (gustavjorlov) [inline src/js/view/components/LogViewer.js]: Dessa hooks ser ut att vara helt onödiga. De används för att sätta `highlightRegExp` tex. varför inte bara använda `props.highlightInput` direkt på rad 39. (samma fråga kring `highlightColor`)

File: src/js/view/reducers/logInfoReducer.js [hunk 2]
```diff
@@ -9,20 +9,17 @@ export const logInfoReducer = (state = initialState, action) => {
       };
     case 'LOGINFO_INCREASE_SIZE': {
```
Comment (gustavjorlov) [inline src/js/view/reducers/logInfoReducer.js]: Missing tests for this very testable file :)

File: src/js/view/components/LogViewer.js (hunk 1 above)
Comment (mewhhaha) [reply src/js/view/components/LogViewer.js]: Det verkar som det fungerar bra att göra som du säger! Vilket ju är underbart. Lite osäker på hur strukturen var innan, men hade något problem med att det blev massuppdateringar för att den räknade om RegExp hela tiden och att LogViewerList uppdaterade om sig konstant (eftersom den har en effekt på regexpförändringar).

File: src/js/view/reducers/logInfoReducer.js (hunk 2 above)
Comment (mewhhaha) [reply src/js/view/reducers/logInfoReducer.js]: Såklart det ska testas! :swimming_man:

**Comment by mewhhaha** [comment]
> * [ ]  Logic in hooks questionable.
> * [ ]  Also, do everyone in the team fully understand hooks?
> * [ ]  Tests for a reducer file.

For the second point, my main reason with this pull request is to see what other people think about using the hooks. My intention was using this as a concrete example of refactoring something that was hard to understand to something that's hopefully easier to understand. I think this wasn't clear in the original post, so I've edited it to make my intentions clearer!

**Pull request merged**